Title       : x11-libs/pixman: upstream patch for X server crashes on AMD64
Assignee    : x11@gentoo.org
Reported    : 2007-10-04 21:18 0000
Updated     : 2007-10-11 09:44:44 0000
Status      : NEW
URL         : https://bugs.freedesktop.org/show_bug.cgi?id=12398
Severity    : major
Priority    : P2
Reporter    : nihilismwow@gmail.com
Product     : Gentoo Linux
Component   : Library
Keywords    : Inclusion
CC          : aballier@gentoo.org
Comments    : 1
Attachments : 0
URL         : http://bugs.gentoo.org/show_bug.cgi?id=194740


Title       : Random crasher in libpixman
Assignee    : sandmann@daimi.au.dk
Reported    : 2007-09-11 22:20 PST
Updated     : 2007-10-12 00:59:50 PST
Status      : NEW
Severity    : major
Priority    : medium
Reporter    : amd@store20.com
Product     : libpixman
Component   : libpixman
CC          : andrea.cimitan@gmail.com
CC          : apitzsch@cs.tu-berlin.de
CC          : brice.goglin@ens-lyon.org
CC          : FJ.Whittle@gmail.com
CC          : hesso@pool.math.tu-berlin.de
CC          : ian.mcdonald@jandi.co.nz
CC          : jan@jgc.homeip.net
CC          : lars@heidieker.de
Comments    : 15
Attachments : 2
URL         : https://bugs.freedesktop.org/show_bug.cgi?id=12398

diff --git a/pixman/pixman-compose.c b/pixman/pixman-compose.c
index 1e91864..027394a 100644
--- a/pixman/pixman-compose.c
+++ b/pixman/pixman-compose.c
@@ -3880,7 +3880,7 @@ static void fbFetchTransformed(bits_image_t * pict, int x, int y, int width, uin
 			    idistx = 256 - distx;
 			    idisty = 256 - disty;
 			    
-			    b = bits + (y1)*stride;
+			    b = bits + (signed)((y1)*stride);
 			    x_off = x1;
 			    
 			    x1_out = (x1 < box.x1) | (x1 >= box.x2);
@@ -3946,7 +3946,7 @@ static void fbFetchTransformed(bits_image_t * pict, int x, int y, int width, uin
 			    idistx = 256 - distx;
 			    idisty = 256 - disty;
 			    
-			    b = bits + (y1)*stride;
+			    b = bits + (signed)((y1)*stride);
 			    x_off = x1;
 			    
 			    tl = pixman_region_contains_point(pict->common.src_clip, x1, y1, &box)
